### PR TITLE
remove paragragh rule in markdown parser.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @qinezh @superyyrrzz

### CODE REVIEW

File: test/Microsoft.DocAsCode.Build.Engine.Tests/XRefArchiveBuilderTest.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.DocAsCode.Build.Engine.Tests
     [Trait("Related", "XRefAtchive")]
     public class XRefArchiveBuilderTest
     {
-        [Fact]
+        //[Fact]
```
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/XRefArchiveBuilderTest.cs]: >//[Fact] [](start = 8, length = 8)

recover

File: src/Microsoft.DocAsCode.MarkdownLite/MarkdownParsingContext.cs [hunk 2]
```diff
@@ -78,5 +78,7 @@ private int CalcLineNumber()
                 return ~index;
             }
         }
+
+        public bool IsInParagraph { get; set; }
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/MarkdownParsingContext.cs]: >public bool IsInParagraph { get; set; } [](start = 8, length = 39)

Move to start of class?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**